pkg/storage/unified/resource/server.go

```diff
@@ -390,6 +391,13 @@ func (s *server) newEvent(ctx context.Context, user claims.AuthInfo, key *Resour
 		if err != nil {
 			return nil, AsErrorResult(err)
 		}
+
+		// restores will restore with a different k8s uid
+		if event.ObjectOld.GetUID() != obj.GetUID() {
+			event.Type = WatchEvent_RESTORED
```

Comment:
i get an error for this new event type via the k8s watch:
```
{"type":"ERROR","object":{"kind":"Dashboard", // etc
```

Reply:
great catch! Moved it to an `added` event